bolt/include/bolt/Core/HashUtilities.h

```diff
@@ -35,6 +36,12 @@ std::string hashBlock(BinaryContext &BC, const BinaryBasicBlock &BB,
 
 std::string hashBlockLoose(BinaryContext &BC, const BinaryBasicBlock &BB);
 
+std::string hashBlockCalls(BinaryContext &BC, const BinaryBasicBlock &BB);
+
+std::string
+hashBlockCalls(const DenseMap<uint32_t, std::string *> &IdToFunctionName,
```

Comment:
how come here you're using pointers to strings rather than references?

Reply:
You're right, references are more appropriate here

Reply:
On second thought after some experimentation, references in maps (both for std and llvm) are a bit tricky to deal with and result in needing particular idioms to not break the map. (https://stackoverflow.com/questions/1543193/why-cant-i-store-references-in-a-stdmap-in-c) Pointers seem to be the better choice

Reply:
Please use StringRef.